### ISSUE TITLE
add smart heater accessory

### ISSUE THREAD

**Issue by itsermo** [milo526/homebridge-tuya-web#26]
i have a tuya-supported smart electric heater that i would love to add to homebridge using this plugin.

i can possibly code the support for it myself, but would need some guidance.  it seems that i would need to add a .js file for a heater in the lib folder, but in addition to that, the BaseAccessory.js must be updated to take into account smart heaters. is that the right place to start?

when i login to the TuyaWeb app on my iPhone, i see that you can turn it on/off, control temperature, enable/disable child lock and set a timer.  for starters, i think on/off and temperature get/set would play nicely with homekit and cover 99% of the cases.  theoretically, i could create a grouped accessory if more switches are found on the device, that automatically populates a switch accessory for child lock, for example.

any guidance on how to get going would be very appreciated. i don't think it should be too hard to do this, but i'm not sure where to begin and how to know what all the API requests are supported by my device.  is there a way to get a printout of this information from the debug??

**Comment by milo526** [comment]
Please read the [README](https://github.com/milo526/homebridge-tuya-web#homebridge-tuya-web), specifically [this section](https://github.com/milo526/homebridge-tuya-web#how-to-check-whether-the-api-this-library-uses-can-control-your-device).

And then create a new issue [following this template](https://github.com/milo526/homebridge-tuya-web/issues/new?assignees=&labels=&template=new-device-request.md&title=%5BNEW+DEVICE%5D+).

If you've followed these steps we can start to discuss this. There is currently no way to know what your device can and can't do; specifically with regards to the HA API that we are using.

In short, if you follow those steps and either the device doesn't show up or the `dev_type` and `ha_type` fields are not one of: `"climate", "cover", "fan", "light", "scene", "switch"` then this will not work and there is nothing we can do about it (except for asking Tuya to start supporting this device in the home assistant  API as also explained in the README and issue template.
